File: KevinBankersen/ql/src/main/java/eu/bankersen/kevin/ql/ast/expr/logic/Not.java
package eu.bankersen.kevin.ql.ast.expr.logic;

import eu.bankersen.kevin.ql.ast.expr.Expr;
import eu.bankersen.kevin.ql.ast.expr.ExprVisitor;
import eu.bankersen.kevin.ql.ast.expr.UnaryExpr;
import eu.bankersen.kevin.ql.ast.object.value.QLValue;
import eu.bankersen.kevin.ql.interpreter.Environment;

public class Not extends UnaryExpr {

    public Not(Expr expr, int line) {
	super(line, expr);
    }

    @Override
    public final QLValue eval(Environment context) {
	return expr().eval(context).not();
    }

    @Override
    public <T, U> T accept(ExprVisitor<T, U> v, U context) {
	return v.visit(this, context);
    }
}

File: Edward Poot/src/ql/ast/expr/unaryoperatorexpr/not.go
package unaryoperatorexpr

import "ql/ast/expr"

type Not struct {
	UnaryOperator
}

func NewNot(value expr.Expr) Not {
	return Not{UnaryOperator{Value: value}}
}

func (n Not) Eval(s interface{}) interface{} {
	return !n.Value.Eval(s).(bool)
}
